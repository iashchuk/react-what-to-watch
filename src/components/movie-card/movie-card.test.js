import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./movie-card.jsx";

const cardData = {
  id: 1,
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  imageSrc: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
};

describe(`MovieCard`, () => {
  it(`renders correctly`, () => {
    const component = renderer
      .create(<MovieCard title={cardData.title} imageSrc={cardData.imageSrc} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  it(`renders correctly with added class`, () => {
    const component = renderer
      .create(
          <MovieCard
            className="catalog__card"
            title={cardData.title}
            imageSrc={cardData.imageSrc}
          />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
