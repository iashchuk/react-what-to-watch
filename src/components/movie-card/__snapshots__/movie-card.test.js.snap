// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders correctly 1`] = `
<article
  className="small-movie-card"
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Fantastic Beasts: The Crimes of Grindelwald"
      height="175"
      src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;

exports[`MovieCard renders correctly with added class 1`] = `
<article
  className="small-movie-card catalog__card"
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Fantastic Beasts: The Crimes of Grindelwald"
      height="175"
      src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
